### ISSUE TITLE
User-configurable overrides refuses span-metrics-size as valid processor

### ISSUE THREAD

**Issue by yvrhdn** [grafana/tempo#3945]
**Describe the bug**
When trying to update `metrics_generator.processors` through the user-configurable overrides, when the array contains the processor `span-metrics-size`, the request is rejected with `metrics_generator.processor "span-metrics-size" is not a known processor, valid values: [service-graphs span-metrics local-blocks]`.

**To Reproduce**
1. Start Tempo (SHA or version)
2. Update `metrics_generator.processors` through the user-configurable overrides

**Expected behavior**
`span-metrics-size` is a valid subprocessor of the span-metrics processor and should be accepted.

**Environment:**
N/A

**Additional Context**

**Comment by vijal-patel** [comment]
@kvrhdn I've started working on this and I'd like to confirm if I should add support for the other spanmetrics sub processors (span-metrics-latency, span-metrics-count) in addition to span-metrics-size

**Comment by kvrhdn** [comment]
Yes that's right. We should also allow all of the subprocessors defined here: https://github.com/grafana/tempo/blob/main/modules/generator/processor/spanmetrics/subprocessors.go
